src/errors.ts

```diff
@@ -0,0 +1,58 @@
+export const APIErrorCodes: Record<string, { name: string; retryable: boolean }> = {
+  '-1': { name: 'InternalSystemError', retryable: true },
+  '2': { name: 'AccessKeyError', retryable: false },
+  '3': { name: 'AuthenticationFailedError', retryable: true },
+  '4': { name: 'InputError', retryable: false },
+  '6': { name: 'DuplicateUsernameError', retryable: false },
+  '9': { name: 'RateLimitError', retryable: true },
+  '16': { name: 'DoesNotExistError', retryable: false },
+  '17': { name: 'NotAllowedError', retryable: false },
+  '18': { name: 'EventNotSupportedError', retryable: false },
+  '19': { name: 'ChannelFeatureNotSupportedError', retryable: false },
+  '20': { name: 'MessageTooLongError', retryable: false },
+  '21': { name: 'MultipleNestingLevelError', retryable: false },
+  '22': { name: 'PayloadTooBigError', retryable: false },
+  '23': { name: 'RequestTimeoutError', retryable: true },
+  '24': { name: 'MaxHeaderSizeExceededError', retryable: false },
+  '40': { name: 'AuthErrorTokenExpired', retryable: false },
+  '41': { name: 'AuthErrorTokenNotValidYet', retryable: false },
+  '42': { name: 'AuthErrorTokenUsedBeforeIssuedAt', retryable: false },
+  '43': { name: 'AuthErrorTokenSignatureInvalid', retryable: false },
+  '44': { name: 'CustomCommandEndpointMissingError', retryable: false },
+  '45': { name: 'CustomCommandEndpointCallError', retryable: true },
+  '46': { name: 'ConnectionIDNotFoundError', retryable: false },
+  '60': { name: 'CoolDownError', retryable: true },
+  '69': { name: 'ErrWrongRegion', retryable: false },
+  '70': { name: 'ErrQueryChannelPermissions', retryable: false },
+  '71': { name: 'ErrTooManyConnections', retryable: true },
+  '99': { name: 'AppSuspendedError', retryable: false },
+};
+
+type APIError = Error & { code?: number; isWSFailure?: boolean };
+
+export function isAPIError(error: Error): error is APIError {
+  return (error as APIError).code !== undefined;
```

Comment:
why in the type definition do we say code is optional to the APIError type, but in the type guard we say an error is only an APIError if it has a code field?

Reply:
good catch, we also need to refactor the way we handle and throw errors in the SDK, worth putting it in the roadmap notion